projects/erp-hgrid/src/components/erp-hierarchical-grid.ts

```diff
@@ -0,0 +1,609 @@
+                    ?sortable=${true}
+                    ?resizable=${true}
+                    visible-when-collapsed=${false}
+                    .formatter="${this.formatDate}">
+                </igc-column>
+            </igc-column-group>
+
+
+            <igc-column-group header="Order Information" ?collapsible="${true}">
+
+                <!-- Show next 4 columns when expanded -->
+                <igc-column
+                    field="orderInformation.country"
+                    header="Country"
+                    width="12%"
+                    ?sortable=${true}
+                    visible-when-collapsed=${false}
+                    .bodyTemplate=${this.countryTemplate}>
+                </igc-column>
+
+                <igc-column
+                    field="orderInformation.city"
+                    header="City"
+                    data-type="string"
+                    width="13%"
+                    ?sortable=${true}
+                    ?resizable=${true}
+                    visible-when-collapsed=${false}>
+                </igc-column>
+                <igc-column
+                    field="orderInformation.zipCode"
+                    header="Zip Code"
+                    data-type="number"
+                    width="9%"
+                    ?sortable=${true}
+                    ?resizable=${true}
+                    visible-when-collapsed=${false}
+                    .formatter="${this.formatNumberAsIs}">
+                </igc-column>
+                <igc-column
+                    field="orderInformation"
+                    header="Address"
+                    data-type="string"
+                    ?sortable=${true}
+                    ?resizable=${true}
+                    visible-when-collapsed=${false}
+                    .formatter="${this.formatAddress}"
+                    .filters="${this.shortAddressFilteringOperand}">
+                </igc-column>
+
+                <!-- Show this column when collapsed -->
+                <igc-column
+                    field="orderInformation"
+                    header="Address"
+                    dataType="string"
+                    ?sortable=${true}
+                    ?resizable=${true}
+                    visible-when-collapsed=${true}
+                    .formatter="${this.formatFullAddress}"
+                    .filters="${this.fullAddressFilteringOperand}">
+                </igc-column>
+
+            </igc-column-group>
+
+          </igc-row-island>
+
+          
+        </igc-hierarchical-grid>
+
+        <!-- Overay shown when hovering on product image -->
+        <div id="product-image-tooltip" role="tooltip">
+          <div class="dialog-header"> ${this.hoveredImageProductName} </div>
+          <div class="dialog-body">
+            <img src="${this.hoveredImageUrl}" alt="${this.hoveredImageProductName}"/>
+          </div>
+        </div>
+       
+      </div>
+    `;
+  }
+
+  static styles = css`
```

Comment:
You should probably move those styles in another file because the file is getting too big . You have two approaches:

Just move this into a separate file like [this](https://github.com/IgniteUI/grid-demos-wc/pull/9/files#diff-4fc605ebb02740088bff90ee1acf44ee239903bbb58ae9bfad32de2ad935b863R1)
Or use scss file and import it like [this](https://github.com/IgniteUI/grid-demos-wc/pull/5/files#diff-56736ab07a823711ba715dacdbb7802e99b2ed9e21b4382d48bfea8bb881dbfeR1). Keep in mind that you must install sass so that vite has a way to preprocess the scss file